Add client net ownership gym (#58)
spatialos/UnrealGDKTestGyms: USER_MANUAL.md

  * Play with 2 clients connected.
  * Each connected client should display a log with "UTask_DelaySimulated::InitSimulatedTask".
  * There should not be logs saying "Error: Simulated task is simulating on the owning client".

##### Client Net Ownership gym
* Demonstrates that:
  * In a zoned environment, setting client net-ownership of an Actor correctly updates the `ComponentPresence` and `EntityACL` components, and allows server RPCs to be sent correctly.
* Contains:
  * A character with a `PlayerController` with key bindings for:
    * (Q) Making the client net-owner for the cube,
    * (R) Sending a server RPC from the client on the cube,
    * (T) Removing the client as the net-owner for the cube.
* To setup the gym:
  * Adjust the settings in `SpatialOS Settings` -> `Editor Settings` -> `Launch` -> `Launch configuration file options` -> `Server Workers` to include `4` servers in a `2x2` grid.
    * Set both `Rectangle grid column count` and `Rectangle grid row count` to 2
    * Set `Instances to launch in editor` to `4`
* To test the gym:
    * Press `Q` to make the client net-owner for the cube.
    * Observe that:

>       * the `SpatialDebugger` authority icon update to the virtual worker ID relating the character.

Reviewer: nit
```suggestion
      * the `SpatialDebugger` authority icon updates to the virtual worker ID relating to the character.
```

>       * in the inspector, the `owner` property on the cube entity is updated to the `PlayerController` and, in the list of component authorities for the cube entity, the `UnrealClientEndpoint` component (ID `9978`) is set to the client worker ID,

Reviewer: nit
```suggestion
      * in the inspector, the `owner` property on the cube entity is updated to the `PlayerController` and, in the list of component authorities for the cube entity, the `UnrealClientEndpoint` component (ID `9978`) is set to the client worker ID.
```

    * Press `R` to send a server RPC from the client on the cube.
    * Observe that:
      * the overhead value above the cube is incremented (this is set by the server and replicated to the client).
    * Press `T` to remove the client as net-owner for the cube.
    * Observe that:

>       * Pressing `R` no longer increments the overhead value,

Reviewer: super nit; inconsistent capitalisation
```suggestion
      * pressing `R` no longer increments the overhead value,
```
are the commas on purpose to make the observation bullet points readable? If so could the bullet points a few lines up use the same? (or all full stops)

>       * in the inspector, the `owner` and `UnrealClientEndpoint` component authority assignment as both unset.

Reviewer: nit: 
```suggestion
      * in the inspector, the `owner` and `UnrealClientEndpoint` component authority assignments are both unset.
```
